code/go/internal/validator/semantic/types.go

```diff
@@ -214,6 +215,9 @@ func readFieldsFolder(fsys fspath.FS, fieldsDir string) ([]string, error) {
 	}
 
 	for _, f := range fs {
+		if filepath.Ext(f.Name()) != ".yml" && filepath.Ext(f.Name()) != ".yaml" {
+			continue
+		}
```

Comment:
I was thinking more in something like this for these validators:
```suggestion
		if filepath.Ext(f.Name()) == ".link" {
			link, err := linkedfiles.New(f.Name())
			if err != nil { ... }
			f, err = os.Stat(link.LinkFilePath)
			if err != nil { ... }
		}
```

Or even better, we could maybe override the `Open()` function in the `fspath.FS` filesystems that validators receive, to make this transparent for them.